packages/hydrogen/src/UIImage.web.tsx

```diff
@@ -0,0 +1,108 @@
+import * as React from 'react';
+
+import { ColorValue, Image as RNImage, ImageProps, StyleSheet, View } from 'react-native';
+import { ColorVariants, useTheme } from './Colors';
+
+export type UIImageProps = ImageProps & {
+    /**
+     * tintColor in some cases don't work on Safary with RNImage.
+     * Here is the issue: https://github.com/necolas/react-native-web/issues/1914
+     * Hence passing this prop we force to use canvas
+     */
+    tintColor?: ColorVariants | null;
+};
+
+/**
+ * Scaling is necessary in order to keep the image quality at the same level
+ */
+const scale = 2;
+
+const TintUIImage = ({ tintColor, ...rest }: UIImageProps) => {
+    const theme = useTheme();
+    const tintColorValue: ColorValue | null = tintColor != null ? theme[tintColor] : null;
+
+    const idRef = React.useRef(Math.random());
```

Comment:
What do you think to use to use sth like `nanoid` to reduce possible collisions?